### DISCUSSION TITLE
undefined reference to `main`

### DISCUSSION THREAD

**Discussion by kevkrist** [NVIDIA/nvbench#183] [Q&A]
I'm running into a problem where the linker is failing with the following error:

/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/Scrt1.o: in function `_start':
(.text+0x1b): undefined reference to `main'
collect2: error: ld returned 1 exit status
gmake[3]: *** [eval/CMakeFiles/eval_probe_nvbench.dir/build.make:136: eval/eval_probe_nvbench] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:635: eval/CMakeFiles/eval_probe_nvbench.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:642: eval/CMakeFiles/eval_probe_nvbench.dir/rule] Error 2
gmake: *** [Makefile:325: eval_probe_nvbench] Error 2

I am using `NVBENCH_BENCH_TYPES` as follows:
`
NVBENCH_BENCH_TYPES(bloom_filter_probe_benchmark,
                    NVBENCH_TYPE_AXES(bits_per_block_list, bits_per_sector_list, z_list, k_list))
  .set_type_axes_names({"B", "S", "Z", "K"})
  .add_int64_power_of_two_axis("N", nvbench::range(10, max_log_num_keys))
  .add_int64_axis("M/N", nvbench::range(4, 10));
`

Is there anything glaringly wrong with how I'm doing this? I'm assuming `NVBENCH_BENCH_TYPES` macro introduces a reference to main somewhere? Which leads me to believe that I'm probably using it incorrectly...

[EDIT] I have a temporary workaround where I simply manually expand the NVBENCH_MAIN() macro. The macro itself doesn't do the trick.

**Comment by alliepiper** [comment]
Can you share a small, complete reproducer?

**Comment by alliepiper** [comment]
Closing due to lack of reproducer.
